Dockerfile

```
@@ -0,0 +1,22 @@
+
+FROM openjdk:21-jdk-slim
+
+ARG MINECRAFT_VERSION=1.21
+ENV SERVER_JAR=/opt/minecraft/server.jar
+WORKDIR /opt/minecraft
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl ca-certificates \
+  && rm -rf /var/lib/apt/lists/*
+
+# Lade automatisch den offiziellen Minecraft-Server herunter
+RUN curl -o $SERVER_JAR https://launcher.mojang.com/v1/objects/95495a7f485eedd84ce928cef5e223b757d2f764/server.jar
```

@A-Marbach 

- Use a build ARG here for `SERVER_JAR_URL` here
- Use curly brackets when you access variables.
- Set curl to fail if the file was not downloaded, or add a check of the file after curl.

This way, you can simplify your `entrypoint.sh`  to just check for the jar file and let the build process download it.

```suggestion
ARG SERVER_JAR_URL=https://launcher.mojang.com/v1/objects/95495a7f485eedd84ce928cef5e223b757d2f764/server.jar
RUN curl -fsSL -o "${SERVER_JAR}" "${SERVER_JAR_URL}"
```

---

Updated Dockerfile: now using ARG SERVER_JAR_URL, curly braces for variables, and curl -fsSL to ensure the file is downloaded. Entrypoint.sh now only checks for the jar file.